test/polymer/behavior-finder_test.js

```diff
@@ -0,0 +1,98 @@
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+"use strict";
+
+const assert = require('chai').assert;
+const fs = require('fs');
+const path = require('path');
+
+const BehaviorDescriptor = require('../../lib/ast/behavior-descriptor').BehaviorDescriptor;
+const BehaviorFinder = require('../../lib/polymer/behavior-finder').BehaviorFinder;
+const JavaScriptParser = require('../../lib/javascript/javascript-parser').JavaScriptParser;
+
+suite('BehaviorFinder', () => {
+
+  let document;
+  let analyzer;
+  let behaviors;
+  let behaviorsList;
+
+  suiteSetup(() => {
+    let parser = new JavaScriptParser();
+    let file = fs.readFileSync(
+        path.resolve(__dirname, '../static/js-behaviors.js'), 'utf8');
+    document = parser.parse(file, '/static/js-behaviors.js');
+    let finder = new BehaviorFinder();
+    let visit = (visitor) => document.visit([visitor]);
+
+    return finder.findEntities(document, visit)
+      .then((entities) => {
+        behaviors = new Map();
+        behaviorsList = entities.filter((e) => e instanceof BehaviorDescriptor);
+        for (let behavior of behaviorsList) {
+          behaviors.set(behavior.is, behavior);
+        }
+      });
+  });
+
+  test('Finds behavior object assignments', () => {
+    assert.equal(behaviorsList.length, 4);
+  });
+
+  test('Supports behaviors at local assignments', () => {
+    assert(behaviors.has('SimpleBehavior'));
+    assert.equal(behaviors.get('SimpleBehavior').properties[0].name, 'simple');
+  });
+
+  test('Supports behaviors with renamed paths', () => {
+    assert(behaviors.has('AwesomeBehavior'));
+    // console.log(analyzer.behaviorsByName['AwesomeBehavior'].properties);
+    var found = false;
+    behaviors.get('AwesomeBehavior').properties.forEach((prop) => {
+      if (prop.name == 'custom') {
+        found = true;
+      }
+    });
+    assert(found);
+  });
+
+  test('Supports behaviors On.Property.Paths', () => {
+    assert(behaviors.has('Really.Really.Deep.Behavior'));
+    assert.equal(
+      behaviors.get('Really.Really.Deep.Behavior').properties[0].name,
+      'deep');
+  });
+
+  test('Supports property array on behaviors', () => {
+    var defaultValue;
+    behaviors.get('AwesomeBehavior').properties.forEach((prop) => {
+      if (prop.name == 'a') {
+        // console.log(prop);
+        defaultValue = prop.default;
+      }
+    });
+    assert.equal(defaultValue, 1);
+  });
+
+  test('Supports chained behaviors', function() {
+    assert(behaviors.has('CustomBehaviorList'));
+    assert.equal(behaviors.get('CustomBehaviorList').behaviors[0], 'SimpleBehavior');
```

Comment:
IMO it's way easier to read:

```typescript
assert.deepEquals(behaviors.get('CustomBehaviorList').behaviors, ['SimpleBehavior', 'CustomNamedBehavior', 'Really.Really.Deep.Behavior']);
```

The error messages are better too (e.g. way easier to read if the results are in a different order)